core/kernel/huk_subkey.c

```diff
@@ -15,17 +16,19 @@ static TEE_Result mac_usage(void *ctx, uint32_t usage)
 }
 
 #ifdef CFG_CORE_HUK_SUBKEY_COMPAT
-/*
- * This gives the result of the default tee_otp_get_die_id()
- * implementation.
- */
-static void get_dummy_die_id(uint8_t *buffer, size_t len)
+static TEE_Result get_otp_die_id(uint8_t *buffer, size_t len)
 {
-	static const char pattern[4] = { 'B', 'E', 'E', 'F' };
-	size_t i;
+	if (IS_ENABLED(CFG_CORE_HUK_SUBKEY_COMPAT_USE_OTP_DIE_ID)) {
+		return tee_otp_get_die_id(buffer, len);
+	} else {
```

Comment:
It seems that checkpatch doesn't like this else since we're returning above, my mistake. I guess we'll need to do it like:
``` C
        static const char pattern...

        if (IS_ENABLED(CFG_CORE_HUK_SUBKEY_COMPAT_USE_OTP_DIE_ID))
                return tee_otp_get_die_id(buffer, len);

        for... 
        return TEE_SUCCESS;
```

Reply:
Sorry, I forgot to run checkpatch before sending the change. My bad :(

Reply:
No worries, that's why we have it in CI. :-)